React/react-assignment/src/pages/Home/Home.tsx

```diff
@@ -7,18 +8,104 @@ type TodoListProps = {
   handleCheckboxToggle: (index: number) => void;
 };
 
+const sortByKeys = [
+  { key: "Title", value: "title" },
+  { key: "Due Date", value: "dueDate" },
+] as const;
+
+const statusKeys = [
+  { key: "Done", value: 1 },
+  { key: "InComplete", value: 0 },
+] as const;
+
 const Home = ({ todoList, handleCheckboxToggle }: TodoListProps) => {
+  const [searchValue, setSearchValue] = useState("");
+  const [todoStatusFilter, setTodoStatusFilter] = useState(-1);
+  const [sortByKey, setSortByKey] = useState("");
+
+  const filteredTodoList = todoList
+    .map((todo, index) => ({ ...todo, originalIndex: index }))
+    .filter((todo) => {
+      const lowerCaseSearchValue = searchValue.toLowerCase();
+      const lowerCaseTodoTitle = todo.title.toLowerCase();
+
+      if (todoStatusFilter === -1) {
+        return lowerCaseTodoTitle.includes(lowerCaseSearchValue);
+      } else {
+        return (
+          Boolean(todoStatusFilter) === todo.isCompleted &&
+          lowerCaseTodoTitle.includes(lowerCaseSearchValue)
+        );
+      }
+    });
+
+  if (sortByKey === "title" || sortByKey === "dueDate") {
+    filteredTodoList.sort((a, b) => {
+      if (a[sortByKey] < b[sortByKey]) return -1;
```

Comment:
cant we use ternary?

Reply:
if statement seems more explicit in this case.